apps/www/_blog/2024-07-01-supabase-realtime-broadcast-and-presence-authorization.mdx

```diff
@@ -0,0 +1,189 @@
+
+<Img
+  src={{
+    dark: '/images/blog/2024-07-01-supabase-realtime-broadcast-and-presence-authorization/schema--dark.png',
+    light:
+      '/images/blog/2024-07-01-supabase-realtime-broadcast-and-presence-authorization/schema--light.png',
+  }}
+  alt="The table that is used to test the policies."
+/>
+
+### One user, one context, one connection
+
+Now that we have set up everything on the database side, let's understand how it works and how we can verify authorization via RLS policies.
+
+Upon channel subscription, we read the contents of your configuration and determine that you want to use a private channel, then we parse the information from your signed JSON Web Token (JWT), load them into a Postgres transaction using `set_config`, and verify them against the new `realtime.messages`table within the same transaction and we store the output as a group of policies within the context of the user's channel process.
+
+<Img
+  src={{
+    dark: '/images/blog/2024-07-01-supabase-realtime-broadcast-and-presence-authorization/one-user-connection--dark.png',
+    light:
+      '/images/blog/2024-07-01-supabase-realtime-broadcast-and-presence-authorization/one-user-connection--light.png',
+  }}
+  alt="Flow of checks done against the databse to determine the policies for a user connection."
+/>
+
+Since we're doing this as soon as the user subscribes to Channel we should see a small latency increase in subscriptions but after that we do not run any queries on the database and we use the stored context to control the actions a user can perform.
+
+Latency between geographically close users is very important for a product like Realtime. To deliver messages as fast as possible between users on our global network, we cache the policies.
+
+We can maintain high throughput and low latency on a Realtime Channel with Broadcast and Presence authorization because:
+
+- the policy is only generated when a user connects to a Channel
+- the policy cached in memory is close to your users
+- the policy is cached for the duration of the connection or until the JWT expires
+
+Another added bonus is the fact that if a user does not have access to a given Channel they won't be able to connect at all and their connections will be rejected.
+
+### Refreshing your Policies
+
+Realtime already has a mechanism to check valid refreshed JWTs used by your connection as a way to ensure the user connected continues to be valid. Now, we've added a step where we re-authorize the user by querying your database's `realtime.messages` table when Realtime receives a new user token. This is to make sure the already connected user continues to be authorized even when you've updated RLS policies.
+
+### Postgres Changes currently not supported
```

Comment:
Let's rephrase this - it sounds like we don't have AuthZ at all, but what you're saying here is that it _is_ supported in a different way

Reply:
changed it slightly on the header and changed the section below it